Objects/abstract.c

```diff
@@ -819,6 +819,21 @@ binary_op(PyObject *v, PyObject *w, const int op_slot, const char *op_name)
     PyObject *result = binary_op1(v, w, op_slot);
     if (result == Py_NotImplemented) {
         Py_DECREF(result);
+
+        if (strcmp(op_name, ">>") == 0 && \
+                strcmp(v->ob_type->tp_name, "builtin_function_or_method") == 0) {
+            PyErr_Format(PyExc_TypeError,
+                 "unsupported operand type(s) for %.100s: "
+                 "'%.100s' and '%.100s'. Did you mean \"print(%s, "
+                 "file={:100!r}).format(%s))\"",
```

Comment:
Don't include an actual format call in the error message - that was just what I used in the issue tracker to represent `PyErr_Format` pllaceholders. Instead, you'll want to use the `%R` format code to invoke `PyObject_Repr` on the operands directly:

    PyErr_Format(PyExc_TypeError,
        "unsupported operand type(s) for %.100s: "
        "'%.100s' and '%.100s'. Did you mean \"print(%.100R, "
        "file={.100R})\"",
        op_name,
        v->ob_type->tp_name,
        w->ob_type->tp_name,
        w, v);

In the general case, you won't be able to recreate the exact text of the original input, as that's no longer available by the time we get here - we only have a reference to the final runtime objects.

We *might* decide it's worthwhile adding a special-case check for the RHS being exactly `PySys_GetObject("stderr")` and displaying `file=sys.stderr` in that case.

Alternatively, we could dispense with the idea of interpolating values entirely, and just use the hardcoded output `"Did you mean \"print(<message>, file=<output_stream>)\""`